mods/dungeon_loot/mapgen.lua

```diff
@@ -0,0 +1,170 @@
+
+	local item_list = dungeon_loot._internal_get_loot(pos.y, dungeontype)
+	-- take random (partial) sample of all possible items
+	assert(#item_list >= dungeon_loot.STACKS_PER_CHEST_MAX)
+	item_list = random_sample(rand, item_list, dungeon_loot.STACKS_PER_CHEST_MAX)
+
+	-- apply chances / randomized amounts and collect resulting items
+	local items = {}
+	for _, loot in ipairs(item_list) do
+		if rand:next(0, 1000) / 1000 <= loot.chance then
+			local itemdef = minetest.registered_items[loot.name]
+			local amount = 1
+			if loot.count ~= nil then
+				amount = rand:next(loot.count[1], loot.count[2])
+			end
+
+			if itemdef.tool_capabilities then
+				for n = 1, amount do
+					local wear = rand:next(0.20 * 65535, 0.75 * 65535) -- 20% to 75% wear
+					table.insert(items, ItemStack({name=loot.name, wear=wear}))
+				end
+			elseif itemdef.stack_max == 1 then
+				-- not stackable, add separately
+				for n = 1, amount do
+					table.insert(items, loot.name)
+				end
+			else
+				table.insert(items, ItemStack({name=loot.name, count=amount}))
+			end
+		end
+	end
+
+	-- place items at random places in chest
+	local inv = minetest.get_meta(pos):get_inventory()
+	local listsz = inv:get_size("main")
+	assert(listsz >= #items)
+	for _, item in ipairs(items) do
+		local index = rand:next(1, listsz)
+		if inv:get_stack("main", index):is_empty() then
+			inv:set_stack("main", index, item)
+		else
+			inv:add_item("main", item) -- space occupied, just put it anywhere
+		end
+	end
+end
+
+
+minetest.register_on_generated(function(minp, maxp, blockseed)
+	local gennotify = minetest.get_mapgen_object("gennotify")
+	local poslist = gennotify["dungeon"] or {}
+	for _, entry in ipairs(gennotify["temple"] or {}) do
+		table.insert(poslist, entry)
+	end
+	if #poslist == 0 then return end
+
+	local noise = minetest.get_perlin(10115, 4, 0.5, 1)
+	local rand = PcgRandom(noise3d_integer(noise, poslist[1]))
+
+	local candidates = {}
+	-- process at most 16 rooms to keep runtime of this predictable
+	local no_process = math.min(#poslist, 16)
+	for i = 1, no_process do
+		local room = find_walls(poslist[i])
+		-- skip small rooms and everything that doesn't at least have 3 walls
+		if math.min(room.size.x, room.size.z) >= 4 and #room.walls >= 3 then
+			table.insert(candidates, room)
+		end
+	end
+
+	local no_chests = rand:next(dungeon_loot.CHESTS_MIN, dungeon_loot.CHESTS_MAX)
+	no_chests = math.min(#candidates, no_chests)
+	local rooms = random_sample(rand, candidates, no_chests)
+
+	for _, room in ipairs(rooms) do
+		-- choose place somewhere in front of any of the walls
+		local wall = room.walls[rand:next(1, #room.walls)]
+		local v, vi -- vector / axis that runs alongside the wall
+		if wall.facing.x ~= 0 then
+			v, vi = {x=0, y=0, z=1}, "z"
+		else
+			v, vi = {x=1, y=0, z=0}, "x"
+		end
+		local chestpos = vector.add(wall.pos, wall.facing)
+		local off = rand:next(-room.size[vi]/2 + 1, room.size[vi]/2 - 1)
```

Comment:
Spaces around '/'.

Reply:
Yes i suspected this, it's a bad habit you have ;] and no excuse. Anyway i won't insist.